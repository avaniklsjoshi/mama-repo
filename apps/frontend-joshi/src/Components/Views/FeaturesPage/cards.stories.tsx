import { ComponentMeta, ComponentStory } from "@storybook/react";
import { BrowserRouter } from "react-router-dom";

import Cards from "./cards";

export default {
  component: Cards,
  title: "Feature Page/Cards",
  parameters: {
    layout: "fullscreen"
  }
} as ComponentMeta<typeof Cards>;
const Template: ComponentStory<typeof Cards> = () => (
  <BrowserRouter>
    <Cards
      cardsList={[
        {
          path: "miscellaneousCoolStuff",
          title: "cardMiscellaneousTitle",
          description: "cardMiscellaneousDescription",
          thumbnail: "../../Assets/Images/launch.png",
          information: "cardMiscellaneousInformation"
        }
      ]}
    />
  </BrowserRouter>
);

export const SelectClick = Template.bind({});
