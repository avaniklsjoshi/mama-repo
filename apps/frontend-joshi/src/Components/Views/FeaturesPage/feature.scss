@import "../../../CommonStyles/variables.scss";

.feature-container {
  margin: 32px;
  .feature-title {
    font-size: $appSubTitleFontSize;
  }
  .feature-cards-conatainer {
    padding: 24px;
    display: flex;
    justify-content: space-around;
    .feature-card {
      margin: 48px;
      .MuiCardHeader-action {
        align-self: center;
      }
      div:first-child {
        padding: 16px;
      }
      .like-share-container {
        justify-content: flex-end;
        display: flex;
      }
    }
  }
}
