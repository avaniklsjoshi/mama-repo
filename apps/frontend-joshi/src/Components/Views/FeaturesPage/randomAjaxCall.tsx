import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import { useDispatch, useSelector } from "react-redux";
import { fetchPostsRequest } from "../../../Redux/Actions/PostsActions/postsActions";
import { RootState } from "../../../Redux/Reducers/rootReducer";

export default function RandomAjaxCall() {
  const dispatch = useDispatch();
  const { t } = useTranslation();

  const { pending, posts, error } = useSelector(
    (state: RootState) => state.posts
  );

  useEffect(() => {
    document.title = t("docTitleNameFeatures");
  });
  useEffect(() => {
    dispatch(fetchPostsRequest());
  }, [dispatch]);

  return (
    <div>
      Features:
      {pending ? (
        <div>Loading...</div>
      ) : error ? (
        <div>Error</div>
      ) : (
        posts?.map((todo, index) => (
          <div key={todo.id}>
            {++index}. {todo.title}
          </div>
        ))
      )}
    </div>
  );
}
