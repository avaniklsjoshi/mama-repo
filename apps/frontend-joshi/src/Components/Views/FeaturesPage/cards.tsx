import "./feature.scss";
import FavoriteIcon from "@mui/icons-material/Favorite";
import ShareIcon from "@mui/icons-material/Share";
import {
  Card,
  CardContent,
  CardHeader,
  CardMedia,
  IconButton,
  Tooltip,
  Typography
} from "@mui/material";
import { InfoIcon } from "../../../Assets/SVG/infoIcon";
import { CARDS_LIST } from "../../../Configs/Constants/const";

export default function Cards() {
  const handleLike = (e: any) => {
    e.stopPropagation();
    console.log("like clicked");
  };
  const handleShare = (e: any) => {
    e.stopPropagation();
    console.log("share clicked");
  };
  const handleInfo = (e: any) => {
    e.stopPropagation();
    console.log("info clicked");
  };
  const handleCard = () => {
    console.log("card clicked");
  };
  return (
    <div className="feature-cards-conatainer">
      {CARDS_LIST.map((card, index) => (
        <div className="feature-card" key={index}>
          <Card sx={{ width: 400 }} onClick={handleCard}>
            <CardHeader
              title={card.title}
              action={
                <Tooltip title={card.information} onClick={handleInfo}>
                  <IconButton>
                    <InfoIcon />
                  </IconButton>
                </Tooltip>
              }
            />
            <CardMedia
              component="img"
              height="346"
              image={card.thumbnail}
              alt={card.description}
            />
            <CardContent>
              <Typography variant="body2" color="text.secondary">
                {card.description}
              </Typography>
            </CardContent>
            <div className="like-share-container">
              <IconButton aria-label="add to favorites" onClick={handleLike}>
                <FavoriteIcon />
              </IconButton>
              <IconButton aria-label="share" onClick={handleShare}>
                <ShareIcon />
              </IconButton>
            </div>
          </Card>
        </div>
      ))}
    </div>
  );
}
