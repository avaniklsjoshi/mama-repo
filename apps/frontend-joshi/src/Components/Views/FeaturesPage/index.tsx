import React, { useEffect } from "react";
import { useTranslation } from "react-i18next";
import "./feature.scss";
import Cards from "./cards";

export default function FeaturesPage() {
  const { t } = useTranslation();

  useEffect(() => {
    document.title = t("docTitleNameFeatures");
  });

  return (
    <div className="feature-container">
      <div className="feature-title">{t("featuresRoute")}</div>
      <Cards />
    </div>
  );
}
