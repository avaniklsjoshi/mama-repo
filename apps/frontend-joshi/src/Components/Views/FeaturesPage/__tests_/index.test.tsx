/* eslint-disable testing-library/no-node-access */
/* eslint-disable testing-library/no-container */
import { act, fireEvent, render, screen } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "styled-components";
import girly from "../../../../Theme/themes/girly";
import Cards from "../cards";
import FeaturesPage from "../index";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: any) => key
  })
}));

describe("Features Page", () => {
  it("renders Features Page", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <BrowserRouter>
          <FeaturesPage />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText("cardMiscellaneousTitle")).toBeInTheDocument();
    expect(
      screen.getByText("cardMiscellaneousDescription")
    ).toBeInTheDocument();
    expect(screen.getByText("cardUITitle")).toBeInTheDocument();
    expect(screen.getByText("cardUIDescription")).toBeInTheDocument();
    expect(screen.getByText("cardReactAndJSTitle")).toBeInTheDocument();
    expect(screen.getByText("cardReactAndJSDescription")).toBeInTheDocument();
  });
  it("renders Features Page Cards", () => {
    const { container } = render(
      <ThemeProvider theme={girly}>
        <BrowserRouter>
          <Cards
            cardsList={[
              {
                path: "miscellaneousCoolStuff",
                title: "cardMiscellaneousTitle",
                description: "cardMiscellaneousDescription",
                thumbnail: "../../Assets/Images/launch.png",
                information: "cardMiscellaneousInformation"
              }
            ]}
          />
        </BrowserRouter>
      </ThemeProvider>
    );
    expect(container).toMatchSnapshot();
    expect(screen.getByText("cardMiscellaneousTitle")).toBeInTheDocument();
    expect(
      screen.getByText("cardMiscellaneousDescription")
    ).toBeInTheDocument();

    const cardButton = screen.getByTestId("feature-card-mui");
    fireEvent.click(cardButton);
    const likeButton = screen.getByTestId("feature-card-like-mui");
    fireEvent.click(likeButton);
    const shareButton = screen.getByTestId("feature-card-share-mui");
    fireEvent.click(shareButton);
    const infoButton = screen.getByTestId("info-card-mui");
    fireEvent.click(infoButton);
  });
});
