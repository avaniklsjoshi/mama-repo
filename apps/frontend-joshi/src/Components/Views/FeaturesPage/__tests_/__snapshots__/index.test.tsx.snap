// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Features Page renders Features Page 1`] = `
<div>
  <div
    class="feature-container"
  >
    <div
      class="bg"
    />
    <div
      class="bg bg2"
    />
    <div
      class="bg bg3"
    />
    <div
      class="feature-cards-conatainer"
    >
      <div
        class="feature-card"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root feature-card-mui css-14cyfy1-MuiPaper-root-MuiCard-root"
          data-testid="feature-card-mui"
          id="miscellaneousCoolStuff"
        >
          <div
            class="MuiCardHeader-root feature-cardheader css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                cardMiscellaneousTitle
              </span>
            </div>
            <div
              class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
            >
              <button
                aria-label="cardMiscellaneousInformation"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                data-testid="info-card-mui"
                tabindex="0"
                type="button"
              >
                <svg
                  height="14px"
                  id="Capa_1"
                  version="1.1"
                  viewBox="0 0 111.577 111.577"
                  width="14px"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <g>
                    <path
                      d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                      fill="#2a272763"
                    />
                  </g>
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="card-image"
          >
            <img
              alt="cardMiscellaneousDescription"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
              height="346"
              src="../../Assets/Images/launch.png"
            />
          </div>
          <div
            class="MuiCardContent-root card-desc css-46bh2p-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              cardMiscellaneousDescription
            </p>
          </div>
          <div
            class="like-share-container"
          >
            <button
              aria-label="add to favorites"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="feature-card-like-mui"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="share"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="feature-card-share-mui"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShareIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="feature-card"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root feature-card-mui css-14cyfy1-MuiPaper-root-MuiCard-root"
          data-testid="feature-card-mui"
          id="uiLearnings"
        >
          <div
            class="MuiCardHeader-root feature-cardheader css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                cardUITitle
              </span>
            </div>
            <div
              class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
            >
              <button
                aria-label="cardUIInformation"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                data-testid="info-card-mui"
                tabindex="0"
                type="button"
              >
                <svg
                  height="14px"
                  id="Capa_1"
                  version="1.1"
                  viewBox="0 0 111.577 111.577"
                  width="14px"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <g>
                    <path
                      d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                      fill="#2a272763"
                    />
                  </g>
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="card-image"
          >
            <img
              alt="cardUIDescription"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
              height="346"
              src="../../Assets/Images/design.png"
            />
          </div>
          <div
            class="MuiCardContent-root card-desc css-46bh2p-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              cardUIDescription
            </p>
          </div>
          <div
            class="like-share-container"
          >
            <button
              aria-label="add to favorites"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="feature-card-like-mui"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="share"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="feature-card-share-mui"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShareIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="feature-card"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root feature-card-mui css-14cyfy1-MuiPaper-root-MuiCard-root"
          data-testid="feature-card-mui"
          id="reactAndJSLearnings"
        >
          <div
            class="MuiCardHeader-root feature-cardheader css-185gdzj-MuiCardHeader-root"
          >
            <div
              class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
            >
              <span
                class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
              >
                cardReactAndJSTitle
              </span>
            </div>
            <div
              class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
            >
              <button
                aria-label="cardReactAndJSInformation"
                class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                data-mui-internal-clone-element="true"
                data-testid="info-card-mui"
                tabindex="0"
                type="button"
              >
                <svg
                  height="14px"
                  id="Capa_1"
                  version="1.1"
                  viewBox="0 0 111.577 111.577"
                  width="14px"
                  x="0px"
                  xml:space="preserve"
                  y="0px"
                >
                  <g>
                    <path
                      d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                      fill="#2a272763"
                    />
                  </g>
                </svg>
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </div>
          </div>
          <div
            class="card-image"
          >
            <img
              alt="cardReactAndJSDescription"
              class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
              height="346"
              src="../../Assets/Images/code.png"
            />
          </div>
          <div
            class="MuiCardContent-root card-desc css-46bh2p-MuiCardContent-root"
          >
            <p
              class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
            >
              cardReactAndJSDescription
            </p>
          </div>
          <div
            class="like-share-container"
          >
            <button
              aria-label="add to favorites"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="feature-card-like-mui"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="FavoriteIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
            <button
              aria-label="share"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-testid="feature-card-share-mui"
              tabindex="0"
              type="button"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ShareIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
                />
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Features Page renders Features Page Cards 1`] = `
<div>
  <div
    class="feature-cards-conatainer"
  >
    <div
      class="feature-card"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root feature-card-mui css-14cyfy1-MuiPaper-root-MuiCard-root"
        data-testid="feature-card-mui"
        id="miscellaneousCoolStuff"
      >
        <div
          class="MuiCardHeader-root feature-cardheader css-185gdzj-MuiCardHeader-root"
        >
          <div
            class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
          >
            <span
              class="MuiTypography-root MuiTypography-h5 MuiCardHeader-title css-1qvr50w-MuiTypography-root"
            >
              cardMiscellaneousTitle
            </span>
          </div>
          <div
            class="MuiCardHeader-action css-sgoict-MuiCardHeader-action"
          >
            <button
              aria-label="cardMiscellaneousInformation"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
              data-mui-internal-clone-element="true"
              data-testid="info-card-mui"
              tabindex="0"
              type="button"
            >
              <svg
                height="14px"
                id="Capa_1"
                version="1.1"
                viewBox="0 0 111.577 111.577"
                width="14px"
                x="0px"
                xml:space="preserve"
                y="0px"
              >
                <g>
                  <path
                    d="M78.962,99.536l-1.559,6.373c-4.677,1.846-8.413,3.251-11.195,4.217c-2.785,0.969-6.021,1.451-9.708,1.451 c-5.662,0-10.066-1.387-13.207-4.142c-3.141-2.766-4.712-6.271-4.712-10.523c0-1.646,0.114-3.339,0.351-5.064 c0.239-1.727,0.619-3.672,1.139-5.846l5.845-20.688c0.52-1.981,0.962-3.858,1.316-5.633c0.359-1.764,0.532-3.387,0.532-4.848 c0-2.642-0.547-4.49-1.636-5.529c-1.089-1.036-3.167-1.562-6.252-1.562c-1.511,0-3.064,0.242-4.647,0.71 c-1.59,0.47-2.949,0.924-4.09,1.346l1.563-6.378c3.829-1.559,7.489-2.894,10.99-4.002c3.501-1.111,6.809-1.667,9.938-1.667 c5.623,0,9.962,1.359,13.009,4.077c3.047,2.72,4.57,6.246,4.57,10.591c0,0.899-0.1,2.483-0.315,4.747 c-0.21,2.269-0.601,4.348-1.171,6.239l-5.82,20.605c-0.477,1.655-0.906,3.547-1.279,5.676c-0.385,2.115-0.569,3.731-0.569,4.815 c0,2.736,0.61,4.604,1.833,5.597c1.232,0.993,3.354,1.487,6.368,1.487c1.415,0,3.025-0.251,4.814-0.744 C76.854,100.348,78.155,99.915,78.962,99.536z M80.438,13.03c0,3.59-1.353,6.656-4.072,9.177c-2.712,2.53-5.98,3.796-9.803,3.796 c-3.835,0-7.111-1.266-9.854-3.796c-2.738-2.522-4.11-5.587-4.11-9.177c0-3.583,1.372-6.654,4.11-9.207 C59.447,1.274,62.729,0,66.563,0c3.822,0,7.091,1.277,9.803,3.823C79.087,6.376,80.438,9.448,80.438,13.03z"
                    fill="#2a272763"
                  />
                </g>
              </svg>
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
        <div
          class="card-image"
        >
          <img
            alt="cardMiscellaneousDescription"
            class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-o69gx8-MuiCardMedia-root"
            height="346"
            src="../../Assets/Images/launch.png"
          />
        </div>
        <div
          class="MuiCardContent-root card-desc css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
          >
            cardMiscellaneousDescription
          </p>
        </div>
        <div
          class="like-share-container"
        >
          <button
            aria-label="add to favorites"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="feature-card-like-mui"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="FavoriteIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 21.35-1.45-1.32C5.4 15.36 2 12.28 2 8.5 2 5.42 4.42 3 7.5 3c1.74 0 3.41.81 4.5 2.09C13.09 3.81 14.76 3 16.5 3 19.58 3 22 5.42 22 8.5c0 3.78-3.4 6.86-8.55 11.54L12 21.35z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            aria-label="share"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
            data-testid="feature-card-share-mui"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ShareIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M18 16.08c-.76 0-1.44.3-1.96.77L8.91 12.7c.05-.23.09-.46.09-.7s-.04-.47-.09-.7l7.05-4.11c.54.5 1.25.81 2.04.81 1.66 0 3-1.34 3-3s-1.34-3-3-3-3 1.34-3 3c0 .24.04.47.09.7L8.04 9.81C7.5 9.31 6.79 9 6 9c-1.66 0-3 1.34-3 3s1.34 3 3 3c.79 0 1.5-.31 2.04-.81l7.12 4.16c-.05.21-.08.43-.08.65 0 1.61 1.31 2.92 2.92 2.92 1.61 0 2.92-1.31 2.92-2.92s-1.31-2.92-2.92-2.92z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
